src/core/common/notifier.hpp

```diff
@@ -0,0 +1,202 @@
+/*
+ *  Copyright (c) 2016-2017, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *  This file defines OpenThread Notifier class.
+ */
+
+#ifndef NOTIFIER_HPP_
+#define NOTIFIER_HPP_
+
+#include "openthread-core-config.h"
+
+#include "utils/wrap_stdint.h"
+#include "utils/wrap_stdbool.h"
+
+#include <openthread/instance.h>
+#include <openthread/types.h>
+
+#include "common/locator.hpp"
+#include "common/tasklet.hpp"
+
+namespace ot {
+
+/**
+ * @addtogroup core-notifier
+ *
+ * @brief
+ *   This module includes definitions for OpenThread Notifier class.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This class implements the OpenThread Notifier.
+ *
+ * It can be used to register callbacks to be notified of state or configuration changes within OpenThread.
+ *
+ */
+class Notifier: public InstanceLocator
+{
+public:
+
+    /**
+     * This class defines a callback instance that can be registered with the `Notifier`.
+     *
+     */
+    class Callback: public OwnerLocator
+    {
+        friend class Notifier;
+
+    public:
+        /**
+         * This type defines the function pointer which is called to notify of state or configuration changes.
+         *
+         * @param[in] aCallback    A reference to callback instance.
+         * @param[in] aFalgs       A bit-field indicating specific state that has changed. See `OT_CHANGED_<STATE>`
```

Comment:
typo: `aFalgs` -> `aFlags`

Reply:
Changed it in new pushed commit. Thanks.